packages/p2p-media-loader-hlsjs/src/services/fragment-loader.ts

```diff
@@ -1,29 +1,56 @@
 import type {
+  FragmentLoaderContext,
+  HlsConfig,
   Loader,
   LoaderCallbacks,
   LoaderConfiguration,
-  LoaderStats,
-  FragmentLoaderContext,
-  HlsConfig,
   LoaderContext,
+  LoaderStats,
 } from "hls.js";
 import { SegmentManager } from "./segment-mananger";
-import { Segment } from "./playlist";
+import { ByteRange, Segment } from "./playlist";
 import Debug from "debug";
 
 export class FragmentLoaderBase implements Loader<FragmentLoaderContext> {
   context!: FragmentLoaderContext;
   config!: LoaderConfiguration | null;
   callbacks!: LoaderCallbacks<FragmentLoaderContext> | null;
   stats: LoaderStats;
-  defaultLoader: Loader<LoaderContext>;
+  createDefaultLoader: () => Loader<LoaderContext>;
+  defaultLoader?: Loader<LoaderContext>;
   segmentManager: SegmentManager;
-  private debug = Debug("p2pml:fragment-loader");
+  response?: { status: number; ok: boolean; url: string; data: ArrayBuffer };
+  abortController: AbortController = new AbortController();
+  private debug = Debug("hls:fragment-loading");
 
   constructor(config: HlsConfig, segmentManager: SegmentManager) {
     this.segmentManager = segmentManager;
-    this.defaultLoader = new config.loader(config);
-    this.stats = this.defaultLoader.stats;
+    this.createDefaultLoader = () => new config.loader(config);
+    this.stats = {
+      aborted: false,
+      chunkCount: 0,
+      loading: { start: 0, first: 0, end: 0 },
+      buffering: new Proxy(
+        { start: 0, first: 0, end: 0, addToStart: 0 },
+        handler
+      ),
+      parsing: new Proxy({ start: 0, end: 0, addToStart: 0 }, handler),
+      total: 0,
+      loaded: 0,
+      bwEstimate: 0,
+      retry: 0,
+    } as any;
```

Comment:
We don't use any. Probably this will work:

```suggestion
    } satisfies LoaderStats;
```